sklearn/metrics/tests/test_classification.py

```diff
@@ -1140,21 +1141,166 @@ def test_multilabel_hamming_loss():
                          hamming_loss, y1, y2, labels=[0, 1])
 
 
-def test_multilabel_jaccard_similarity_score():
+def test_jaccard_score_validation():
+    y_true = np.array([0, 1, 0, 1, 1])
+    y_pred = np.array([0, 1, 0, 1, 1])
+    assert_raise_message(ValueError, "pos_label=2 is not a valid label: "
+                         "array([0, 1])", jaccard_score, y_true,
+                         y_pred, average='binary', pos_label=2)
+
+    y_true = np.array([[0, 1, 1], [1, 0, 0]])
+    y_pred = np.array([[1, 1, 1], [1, 0, 1]])
+    msg1 = ("Target is multilabel-indicator but average='binary'. "
+            "Please choose another average setting.")
+    assert_raise_message(ValueError, msg1, jaccard_score, y_true,
+                         y_pred, average='binary', pos_label=-1)
+
+    y_true = np.array([0, 1, 1, 0, 2])
+    y_pred = np.array([1, 1, 1, 1, 0])
+    msg2 = ("Target is multiclass but average='binary'. Please choose "
+            "another average setting.")
```

Comment:
Same remark:

Maybe we should be extra helpful and suggest the valid choices for this kind of problem ("macro", "micro", "weighted" or None).